src/token.rs

```diff
@@ -1,7 +1,36 @@
+use std::ops::Range;
+
+//TODO: Source file
 #[derive(Clone, Copy, PartialEq, Debug)]
 pub struct Location {
-    pub line: u32,
-    pub pos: u32,
+    /// Byte offset into the source string where the first char begins
+    pub start: u32,
+    /// Byte offset into the source string where the last char ends
```

Comment:
nit: Maybe mention that it is exclusive, even if that's how Rust ranges usually are.